### ISSUE TITLE
wrong virtcol("']") if last yanked character is non-ASCII

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vim/vim#277]
```
Assume you have a buffer with exactly two lines. The first line containing 
exactly one character, e.g. U+2502 (│), the second line containing exactly 
one character, e.g. U+0041 (a):

line 1: │
line 2: a

What steps will reproduce the problem?
1. Place the cursor at line 1, virtcol 1
2. Start visual mode and yank one character and echo the virtcol of the last 
yanked part, i.e. type:
vy:echo virtcol("']")
The answer will be 2, although line 1 has no second virt-column.

For the ASCII-character you get a different answer: Doing the same thing with 
line 2, virtcol 1
vy:echo virtcol("']")
will give as answer 1. 

What is the expected output? What do you see instead?
Expected output: in both cases "1", because there is only one virtual column in 
every line. Instead I get virtcol 2 in the first line.

What version of the product are you using? On what operating system?
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Oct 18 2014 23:29:48)
Included patches: 1-207, 209-481
OS: Fedora 20



```

Original issue reported on code.google.com by `Ludwi...@gmx.de` on 30 Oct 2014 at 11:37

**Comment by luchr** [comment]
Here is an update of this/my very old bugreport:

| Line | Codepoints           | yanked VISUAL cols   | output of (*)|
|------|----------------------|----------------------|--------------|
| aâ   | U+0061 U+0061 U+0302 |   [1,2]              | [1,3]        |
| â    | U+0061 U+0302        |   [1,1]              | [1,2]        |
| ─    | U+2500               |   [1,1]              | [1,2]        |
| 😀a  | U+01F600 U+0061      |   [1,3]              | [2,3]        |
| a😀  | U+0061 U+01F600      |   [1,3]              | [1,4]        |

(*) ` echo [virtcol("'["),virtcol("']")] `

Why? 
see `charset.c`, function `getvcol`; especially how the case `ptr` is pointing to the character (extended grapheme cluster) directly before `posptr` is handled.

**Comment by brammool** [comment]
This was fixed in patch 8.0.0250

**Comment by lacygoill** [comment]
Since the issue has been fixed, then this todo item is no longer relevant:

https://github.com/vim/vim/blob/21829c5f2c86cd525c8468121b4fc54c5d75bf6e/runtime/doc/todo.txt#L1643-L1644
